src/verifier.hpp

```diff
@@ -111,8 +111,8 @@ class Verifier {
                 new_ys.push_back(std::get<0>(results));
                 new_metadata.push_back(std::get<1>(results));
             }
-            for (size_t i = 0; i < new_ys.size(); i++) {
-                if (new_ys[i].GetSize() <= 0) {
+            for (auto & new_y : new_ys) {
```

Comment:
I think `auto const&` would be nicer here, to clarify that this loop doesn't mutate the elements